petclinic-frontend/src/features/products/ProductsList.tsx

```diff
@@ -82,6 +85,26 @@ export default function ProductList(): JSX.Element {
     setMaxPrice(undefined);
   };
 
+  const handleProductClick = (product: ProductModel): void => {
+    setRecentlyClickedProducts(listOfProducts => {
+      const updatedProducts = [];
+
+      for (const p of listOfProducts) {
+        if (p.productId !== product.productId) {
+          updatedProducts.push(p);
+        }
+      }
```

Comment:
```suggestion
      for (const currentProduct of listOfProducts) {
        if (currentProduct.productId !== product.productId) {
          updatedProducts.push(currentProduct);
        }
      }
```

maybe consider making it more readable with changing "p" to currentProduct

Reply:
Good point! I've updated p to currentProduct to make the code more readable. Thanks for the suggestion!